test/clienthellotest.c

```diff
@@ -95,55 +95,47 @@ static int test_client_hello(int currtest)
          * F5_WORKAROUND_MIN_MSG_LEN bytes long - meaning padding will be
          * needed. Also add some dummy ALPN protocols in case we still don't
          * have enough.
-         * In the padding not needed case we assume the test will pass, but then
-         * set testresult to 0 if we see the padding extension.
          */
         if (currtest == TEST_ADD_PADDING
-                && (!SSL_CTX_set_cipher_list(ctx, "ALL")
-                    || SSL_CTX_set_alpn_protos(ctx,
+                && (!TEST_true(SSL_CTX_set_cipher_list(ctx, "ALL"))
+                    || !TEST_false(SSL_CTX_set_alpn_protos(ctx,
                                                (unsigned char *)alpn_prots,
-                                               sizeof(alpn_prots) - 1)))
+                                               sizeof(alpn_prots) - 1))))
             goto end;
-        else if (currtest == TEST_PADDING_NOT_NEEDED)
-            testresult = 1;
+
         break;
 
     default:
         goto end;
     }
 
     con = SSL_new(ctx);
-    if (con == NULL)
+    if (!TEST_ptr(con))
         goto end;
 
     if (currtest == TEST_ADD_PADDING_AND_PSK) {
         sessbio = BIO_new_file(sessionfile, "r");
-        if (sessbio == NULL) {
-            printf("Unable to open session.pem\n");
+        if (!TEST_ptr(sessbio)) {
+            TEST_info("Unable to open session.pem");
             goto end;
         }
         sess = PEM_read_bio_SSL_SESSION(sessbio, NULL, NULL, NULL);
-        if (sess == NULL) {
-            printf("Unable to load SSL_SESSION\n");
+        if (TEST_ptr(sess)) {
```

Comment:
Good catch. I couldn't figure out for a while why this mistake didn't cause the test to fail. It's because this test is only run if TLSv1.3 is enabled - and I didn't enable it locally.